scripts/scraping/textage.mjs

```diff
@@ -0,0 +1,119 @@
+const textageDir = path.join(__dirname, "textage")
+console.log(textageDir)
+
+async function exists(f) {
+  try {
+    await fs.promises.stat(f);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+export async function textageDL(force = false) {
+  const textageScrapeReady = await Promise.all(textageFiles.map((fn) => exists(`${textageDir}/${fn}.js`))).then((a) => a.every((v) => (v)));
+  if (force || !textageScrapeReady) {
+    console.log("Redownloading source JS from textage...")
+    // Clear out the existing textage JS, if it exists.
+    if (exists(textageDir)) {
+      await fs.promises.rm(textageDir, {"recursive": true, "force": true})
+    }
+    // Redownload all the necessary textage JS.
+    await fs.promises.mkdir(textageDir).catch(()=>{})
+    
+    for (let fn of textageFiles) {
+      if (await exists(`${textageDir}/${fn}.js`)) {
+        console.log(`Don't need to redownload ${fn}`)
+        continue;
+      }
+      console.log(`Downloading ${fn}...`)
+      
+      let req = new Axios({
+        method: 'get',
+        url: `https://textage.cc/score/${fn}.js`,
+        responseType: 'stream'
+      })
+      await req.get(`https://textage.cc/score/${fn}.js`).then(function (response) {
+        const writer = fs.createWriteStream(path.join(textageDir, `${fn}.js`))
+        response.data.pipe(iconv.decodeStream("shift-jis"))
+                     .pipe(iconv.encodeStream("utf-8"))
+                     .pipe(writer)
+        return new Promise((resolve, reject) => {
+          writer.on('error', reject);
+          response.data.on('end', resolve)
+        })
+      });
+    }
+
+    // Double-check that we got all of the textage JS.
+    const textageScrapeSuccess = textageFiles.every((fn) => exists(`scraping/textage/${fn}.js`));
+    if (!textageScrapeSuccess) {
+      console.log(
+        `Failed to download textage JS sources. Invoke like 'yarn import:iidx [rescrape]'`,
+      );
+    }
+    return textageScrapeSuccess;
+  }
+  else {
+    console.log("Not redownloading source JS from textage")
+    return textageScrapeReady;
+  }
+}
+
+export async function fakeTextage(force = false) {
+  // https://stackoverflow.com/questions/950087/how-do-i-include-a-javascript-file-in-another-javascript-file
+
+  const chartSlot = ["inclusion", "SPB", "SPN", "SPH", "SPA", "SPL", "DPB", "DPN", "DPH", "DPA", "DPL"];
+   
+  var dom = new JSDOM('<!DOCTYPE html><head><meta charset="UTF-8"></head>', {runScripts: "dangerously", resources: "usable"});
+  var document = dom.window.document;
+
+  await textageDL(force);
+
+  for (let fn of textageFiles) {
+    let fnLoader = function(doc) {return new Promise(function(resolve) {
+      let script = doc.createElement("script");
+      script.type = "text/javascript";
+      script.charset = "UTF-8";
+      script.src = "file:///" + path.join(textageDir, `${fn}.js`);
+      script.async = false;
+      script.onload = function() {resolve(doc);};
+
+      doc.head.appendChild(script);
```

Comment:
I went down that route partially because I tried and failed to just include them directly in the import script 😤 after noticing that the DDR ingest uses JSDOM, I poked around to see what else could be done with that, and found an extremely relevant stackoverflow post hah